File: src/main/webapp/resources/js/services/CategoryService.js
/* global _contextPath */

storezillaadminapp.service('CategoryService',function($http){
    
   var listCategories = [];

    this.getAllCategories = function() { 
        return $http.get(_contextPath+'/categories');
    };

    this.getCategoryById = function(categoryId) { 
        alert(_contextPath+'/categories/edit/'+categoryId);
        return $http.get(_contextPath+'/categories/edit/'+categoryId);
    };
    
    this.addCategory = function(category) {
      return $http.post(_contextPath+'/categories/add',category);  
    };
    
    this.updateCategory = function(category) {
      return $http.put(_contextPath+'/categories/update',category);  
    };

    this.removeCategory = function(id) {
        return $http.delete(_contextPath+'/categories/remove/'+id);
    };
});

File: src/main/java/org/storezilla/category/service/CategoryService.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.storezilla.category.service;

import java.util.List;
import org.storezilla.category.model.Category;

/**
 *
 * @author LPT466
 */
public interface CategoryService {
    public void addCategory(Category category);
    public void removeCategory(int categoryId);
    public List<Category> listCategories();
    public Category getCategoryById(int categoryId);
    public void updateCategory(Category category);
}
